newIDE/app/src/EventsSheet/EventsTree/helpers.js

```diff
@@ -0,0 +1,92 @@
+import { type SortableTreeNode } from '.';
+
+export type MoveFunctionArguments = {
+  targetNode: SortableTreeNode,
+  node: SortableTreeNode,
+};
+
+export const moveEventToEventsList = ({
+  targetEventsList,
+  movingEvent,
+  initialEventsList,
+  toIndex,
+}: {
+  targetEventsList: gdEventsList,
+  movingEvent: ?gdBaseEvent,
+  initialEventsList: gdEventsList,
+  toIndex: number,
+}) => {
+  if (!movingEvent) return;
+  initialEventsList.moveEventToAnotherEventsList(
+    movingEvent,
+    targetEventsList,
+    toIndex
+  );
+};
+
+export const moveNodeAsSubevent = ({
+  targetNode,
+  node,
+}: MoveFunctionArguments) => {
+  if (!targetNode.event) return;
+  moveEventToEventsList({
+    targetEventsList: targetNode.event.getSubEvents(),
+    movingEvent: node.event,
+    initialEventsList: node.eventsList,
+    toIndex: 0,
+  });
+};
+
+export const moveNodeBelow = ({ targetNode, node }: MoveFunctionArguments) => {
+  if (!targetNode.event) return;
+  const toIndex =
+    node.indexInList < targetNode.indexInList && node.depth === targetNode.depth
+      ? targetNode.indexInList
+      : targetNode.indexInList + 1;
+  moveEventToEventsList({
+    targetEventsList: targetNode.eventsList,
+    movingEvent: node.event,
+    initialEventsList: node.eventsList,
+    toIndex,
+  });
+};
+
+export const moveNodeAbove = ({ targetNode, node }: MoveFunctionArguments) => {
+  const toIndex =
+    node.indexInList < targetNode.indexInList && node.depth === targetNode.depth
+      ? targetNode.indexInList - 1
+      : targetNode.indexInList;
+  moveEventToEventsList({
+    targetEventsList: targetNode.eventsList,
+    movingEvent: node.event,
+    initialEventsList: node.eventsList,
+    toIndex,
+  });
+};
+
+// Optimises react-sortable-tree isDescendant that processes every children with a .some method.
```

Comment:
Is the comment up to date? I don't see a `.some` in the code

Reply:
I actually browsed react-sortable-tree's utils function and was talking about their "isDescendant" method.
https://github.com/frontend-collective/react-sortable-tree/blob/52f0bcf51a80297ab63d5d1cfb0b32f4e3f0e702/src/utils/tree-data-utils.js#L1032

Reply:
I don't know if it's worth adding this comment. You're saying that you could have used RST's isDescendant method but you preferred a more optimized version of your own. But I don't know what to do with this info: you're not overriding a RST default method or using it as an unexpected props for a component.